### PR TITLE
ICU-21569 LSTM Part 1- data file and build tool

### PR DESCRIPTION
**Pull request by FrankYFTang**
NOT including the building of LSTM files by default
Allow developer to use filter to include LSTM

##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21569
- [X] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by FrankYFTang** [comment]
The lstm files are generated from 
https://github.com/unicode-org/lstm_word_segmentation/blob/master/convert_lstm_model.py

File: icu4c/source/data/BUILDRULES.py [hunk 1]
```diff
@@ -457,6 +458,35 @@ def generate_translit(config, io, common_vars):
     ]
 
 
+def generate_brkitr_lstm(config, io, common_vars):
+    input_files = [InFile(filename) for filename in io.glob("brkitr/lstm/*.txt")]
+    dep_files = set(InFile(filename) for filename in io.glob("brkitr/lstm/*.txt"))
+    dep_files -= set(input_files)
+    dep_files = list(sorted(dep_files))
```
Comment (sffc) [inline icu4c/source/data/BUILDRULES.py:465]: Suggestion (optional): I think `dep_files` ends up being empty.  You could delete it and pass an empty array for `dep_targets` down below in the RepeatedOrSingleExecutionRequest.

**Review by sffc — APPROVED** [review]

File: icu4c/source/data/BUILDRULES.py (hunk 1 above)
Comment (FrankYFTang) [reply icu4c/source/data/BUILDRULES.py]: DONE

**Comment by FrankYFTang** [comment]
PTAL

**Review by sffc — APPROVED** [review]
LGTM from me, but I'll flag @markusicu and @srl295 as an FYI that this PR adds quite a bit of size to the ICU repository (not the release, just the repo, and probably icu-data.zip as well).

**Comment by srl295** [comment]
Thought it's not in by default. What's the size delta ?

**Comment by FrankYFTang** [comment]
> Thought it's not in by default. What's the size delta ?

What size are you talking about?

**Comment by FrankYFTang** [comment]
The reason the size delta of .dat should be 0 (or maybe +30 bytes) is because all the lstm model by default won't be build. It will only be build if the filter file explicitly add them.

**Comment by srl295** [comment]
Right, mis read. I mean what is the delta repo size?

**Comment by FrankYFTang** [comment]
I think I already answer you the rough number above
> If you are talking about the size impact to the source data files in the source code- +5.7 M
> Just about ~30 lines changes of the pre-exist file and one .h/.cpp file which total about 900 lines.

**Comment by srl295** [comment]
Sounds fine

**Pull request merged**